lib/handlers/authorize-handler.js

```diff
@@ -81,77 +75,64 @@ AuthorizeHandler.prototype.handle = function(request, response) {
   }
 
   var fns = [
-    this.generateAuthorizationCode(),
-    this.getAuthorizationCodeLifetime(),
-    this.getClient(request),
+    this.getClient(request, response),
     this.getUser(request, response)
   ];
 
   return Promise.all(fns)
     .bind(this)
-    .spread(function(authorizationCode, expiresAt, client, user) {
-      var uri = this.getRedirectUri(request, client);
-      var scope;
-      var state;
-
+    .spread(function (client, user) {
       return Promise.bind(this)
-        .then(function() {
-          scope = this.getScope(request);
-          state = this.getState(request);
-
-          return this.saveAuthorizationCode(authorizationCode, expiresAt, scope, client, uri, user);
-        })
-        .then(function(code) {
-          var responseType = this.getResponseType(request, code);
-          var redirectUri = this.buildSuccessRedirectUri(uri, responseType);
-
-          this.updateResponse(response, redirectUri, state);
-
-          return code;
+        .then(function () {
+          return this.handleResponseType(request, client, user);
         })
-        .catch(function(e) {
-          if (!(e instanceof OAuthError)) {
-            e = new ServerError(e);
-          }
-
-          var redirectUri = this.buildErrorRedirectUri(uri, e);
+        .then(function ({ redirectUri, result }) {
+          this.updateResponse(response, redirectUri);
 
-          this.updateResponse(response, redirectUri, state);
-
-          throw e;
```

Comment:
This behavior was removed, is this intentional?

Reply:
Yes, this behaviour was related to `code-response-types` and so it was moved into [code-response-types.js](https://github.com/uphold-forks/node-oauth2-server/pull/1/files#diff-86d4ccac89f88d5237c96d7bf65facedR93).